### PR TITLE
.NET Core 2.1 runtime

### LINKED ISSUE #6: .gitignore
Hi all,

Shouldn't we add a `.gitignore` file to avoid runtime specific artifacts to be check in?
eg: Java .classes, .NET .dll and so on.

### PR DESCRIPTION
**Pull request by allantargino**
fixes kubeless/kubeless#942
fixes #6 

This refactor includes the major changes:
* Updates runtime version to .NET Core 2.1;
* Removes legacy code (when it used to need compilation inside the runtime - not as a kubeless phase);
* Uses `ILogger` instead of `Console.Write` - adding more util information to logs;
* Adds unit tests to `Kubeless.Core` (covering basic invoking and parameter handling);
* Adds `.gitignore` for .NET Core artifacts;

Tasks:
- [X] Code refactor
- [x] Build new image
- [x] Update `dotnetcore.jsonnet` to `dotnetcore2.1`
- [x] Ready to review

### CODE REVIEW

**Comment by allantargino** [comment]
@andresmgot , any idea why the CircleCI was not triggered?

**Comment by andresmgot** [comment]
Now it got triggered, there was a missing configuration in CircleCI :)

**Comment by andresmgot** [comment]
@allantargino You need to update the `Makefile` of the `dotnetcore` folder in order to adapt the tests.

Also, can we maintain the previous entry for `dotnetcore2.0` in the Jsonnet file? Just to avoid breaking changes for people already using it. We can tell people that they should migrate to `2.1` and that the `2.0` will disappear but giving them some time to migrate.

**Comment by allantargino** [comment]
Thanks for fixing it :)
Yes, it totally makes sense to keep compatibility for some time - I'm keeping both for now.

File: stable/dotnetcore/Makefile [hunk 1]
```diff
@@ -3,22 +3,22 @@ deploy: get-dotnetcore get-dotnetcore-dependency
 test: get-dotnetcore-verify get-dotnetcore-dependency-verify
```
Comment (andresmgot) [inline stable/dotnetcore/Makefile]: just to ensure we don't break 2.0, could you add two cases `get-dotnetcore20` and `get-dotnetcore20-verify`? (using the previous runtime)

Comment (allantargino) [reply stable/dotnetcore/Makefile]: Good point! Gonna fix :)

**Pull request merged**